src/components/Table/GridTable.stories.tsx

```diff
@@ -1162,6 +1162,98 @@ export function SelectableRows() {
   );
 }
 
+export function SelectableChildrenRows() {
+  type HeaderRow = { kind: "header"; id: string; data: undefined };
+  type ParentRow = { kind: "parent"; id: string; data: { name: string } };
+  type ChildRow = { kind: "child"; id: string; data: { name: string } };
+  type GrandChildRow = { kind: "grandChild"; id: string; data: { name: string } };
+  type Row = HeaderRow | ParentRow | ChildRow | GrandChildRow;
+
+  const selectCol = selectColumn<Row>();
+
+  const nameCol: GridColumn<Row> = {
+    header: "Name",
+    parent: ({ name }) => name,
+    child: ({ name }) => name,
+    grandChild: ({ name }) => name,
+    mw: "160px",
+  };
+
+  return (
+    <>
+      <GridTable
+        columns={[collapseColumn<Row>(), selectCol, nameCol]}
+        style={{ rowHeight: "fixed" }}
+        rows={
+          [
+            simpleHeader,
+            {
+              kind: "parent",
+              id: "1",
+              data: { name: "Howard Stark" },
+              initSelected: true,
+              inferSelectedState: false,
```

Comment:
Afaict we never set `inferSelectedState` to true? Is it really necessary for the test? Wondering if we could just delete it.

Reply:
Hi @stephenh 

That is the idea, that  `inferSelectedState` don't affect the selection of the child and grandchild if the parent is set or not